### PR TITLE
Add filename into .md5 file

### PR DESCRIPTION
**Pull request by mgigante**
Please can you take a look to this fix. Generated .md5 files do not work with my kodi 17.1 installation. The fix adds filename into .md5 file.

### CODE REVIEW

**Comment by chadparry** [comment]
Thanks for the fix! I'm only running 17.0, so I didn't test this out. It looks like the most accurate fix would be to have an asterisk instead of a space in front of the name. That indicates a binary file.
```python
        sig.write('{} *{}'.format(checksum.hexdigest(), os.path.basename(archive_path)))
```
Would you have time to test whether the above change works as well for you?

**Comment by mgigante** [comment]
Yes, I can test the changes. I also have and additional fix on whole repository addons.xml.md5 generation. Will do an additional pull request.
Thanks

**Comment by mgigante** [comment]
Yes, the fix you proposed works fine.
I made an additional commit for global addons.xml.md5 generation to the pull request. Please take a look at it.

File: tools/create_repository.py [hunk 1]
```diff
@@ -335,10 +335,7 @@ def create_repository(
         info_file.write(info_contents)
 
     # Calculate the signature.
-    digest = hashlib.md5(info_contents).hexdigest()
-    with open(checksum_path, 'w') as sig:
-        sig.write(digest)
-
+    generate_checksum(info_path)
```
Comment (chadparry) [inline tools/create_repository.py]: The checksum path is an argument to the script, and this would ignore it.

File: tools/create_repository.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def generate_checksum(archive_path):
         for chunk in iter(lambda: archive_contents.read(4096), b''):
             checksum.update(chunk)
     with open(checksum_path, 'w') as sig:
-        sig.write(checksum.hexdigest())
+        sig.write('{} *{}'.format(checksum.hexdigest(), os.path.basename(archive_path)))
```
Comment (chadparry) [inline tools/create_repository.py]: I should probably also add a `'\n'` to the end of the line.

File: tools/create_repository.py (hunk 1 above)
Comment (chadparry) [inline tools/create_repository.py]: I think the docs say that the checksum should be for the raw addon.xml contents, not the zipped addon.xml.gz contents. That means that info_contents should be used as the input to hashlib, not the info_path file.

**Comment by chadparry** [comment]
There are a couple small nits I am going to change, but there's no reason for any of them to be blocking. I'll merge this request and have a new release out in a bit.

**Comment by chadparry** [comment]
Oh, I spoke too soon. I guess this pull request was made against master, when it needs to go into tools-develop. I merge into master from the develop and tools-develop branch when an upgrade is ready. So instead of accepting this pull request as-is, I'll merge your changes manually.

File: tools/create_repository.py (hunk 1 above)
Comment (chadparry) [inline tools/create_repository.py]: Looks like I remembered this wrong. I don't need info_contents. I'll definitely be keeping your change to this part of the file.

**Comment by chadparry** [comment]
All done. Thanks for the fix, @mgigante!

**Comment by mgigante** [comment]
Thank you for all the changes. Sorry I couldn't follow the thread during the week-end.

**Pull request merged**